### ISSUE TITLE
Screen won't lock when closing laptop lid under certain circumstances.

### ISSUE THREAD

**Issue by aly199** [linuxmint/cinnamon#9685]
```
 * Cinnamon version (cinnamon --version)
   - Please specify if you are using the daily builds PPA
     (https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds).
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
 * Graphics hardware *and* driver used
 * 32 or 64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**
Screen won't lock when laptop lid is closed, if you click a systray icon that pops up a menu (such as network manager, power, volume settings).

**Steps to reproduce**
Click the network manager icon.
With the pop-up still open, close the laptop lid.
Wait for computer to suspend.
Open the laptop lid.
Screen won't be locked.

**Expected behaviour**
Screen should be locked when laptop returns from suspended state.

**Other information**
Also, if you hit Ctrl+Alt+L to lock the screen while those pop-ups are showing, the shortcut won't lock the screen either.

**Configuration info, if applicable**
If this bug report is related to an Applet, Desklet or Extension, please paste (or use a pastebin service) the offending extension's settings, if it has any.  You can obtain this by opening its configuration, clicking the 'hamburger' button in the upper-right corner of the window, and selecting "Export to a file". **Please be sure to review the contents and remove any personal data it may contain.**

```

Cinnamon 4.6.7
Linux Mint 20 Cinnamon
Graphics Intel Corporation HD Graphics 620
64 bit

```

**Comment by Krafting** [comment]
this is an issue that have been here for a loooong time, and still not fixed, see #1430 
This should be worked on and fixed.

**Comment by leigh123linux** [comment]
Duplicate of https://github.com/linuxmint/cinnamon/issues/1430
